python/lib/dependabot/python/file_parser/setup_file_parser.rb

```diff
@@ -62,6 +69,7 @@ def parsed_setup_file
             requirements = SharedHelpers.run_helper_subprocess(
               command: "pyenv exec python #{NativeHelpers.python_helper_path}",
               function: "parse_setup",
+              env: { "PYENV_VERSION" => SUBPROCESS_PYTHON },
```

Comment:
Not sure I follow this, is this equivalent to `pyenv local #{SUBPROCESS_PYTHON}`? Think we'll need to detect the installed python version and use this instead of the latest version.

Unrelated to this PR, it looks like the logic for detecting and setting the python version is duplicated in subtly different ways across the codebase. Keen to understand how this works before merging this.

Reply:
It's one way of ensuring that the pyenv exec command above is actually executed with is the same as the one the dependabot already has installed by default, as opposed to the one it tries to determine from the repo itself. I _think_ this is an okay thing to do, since the function that is being called is one controlled on the dependabot side (parse_setup)